hyperspy/events.py

```diff
@@ -54,16 +103,51 @@ def suppress(self):
             obj.val_b = b
         obj.events.values_changed.trigger()
         """
-        return EventSuppressionContext(self)
+        return EventsSuppressionContext(self)
 
 
 class Event(object):
 
     def __init__(self):
         self._connected = {}
-        self.suppress = False
+        self._suppress = False
+
+    def suppress(self, function=None):
```

Comment:
Note that `Events.suppress()` was turned into a function, as that corresponds best with the typical use pattern of context managers. `Event.suppress()` would therefore be a function, even without the optional argument.